iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/KubernetesSensor.java

```diff
@@ -115,6 +115,13 @@ protected String getActivationSettingKey() {
 
   @Override
   protected FilePredicate mainFilePredicate(SensorContext sensorContext) {
+    FilePredicates predicates = sensorContext.fileSystem().predicates();
+    return predicates.or(
```

Comment:
I would suggest to restructure this a bit:
* `mainFilePredicate` is `and(hasType(MAIN), customFilePredicate)`
* `customFilePredicate` is `or(yamlK8sOrHelmFilePredicate, tplHelmFilePredicate)`

I think this would align with the original design of this class better

Reply:
Good idea